### DISCUSSION TITLE
Support AWS and CSI velero Plugin to ARM and ARM64

### DISCUSSION THREAD

**Discussion by i5Js** [ricsanfre/pi-cluster#111] [Q&A] [answered]
Hello Ricardo, 

It's not an issue, but a question. I would like to add the backup and restore to my k3s cluster. Like you I am using raspberry, but mixed arm and arm64. I have tried to build the plugins for those architectures, but I was not able to find the way. Can you help me?
We can talk in Spanish ;)

Gracias!

**Comment by ricsanfre** [comment, marked as answer]
Hola @i5Js,

velero-plugin-for-aws is already supporting arm, isn't it?

velero-plugin-for-csi currently only supports amd64 architecture. There is an open issue since Oct 21 to provide support for arm64 architecture.

A PR, developed by @blampe, were submitted in Aug 22 to fix this issue, but it seems that the community does not have the resources to test using ARM architectures and it has never been merged into the main branch.

Check out https://github.com/ricsanfre/pi-cluster/issues/31#issuecomment-1272371364 to find issue # and its associated PR in velero repository.

To test velero CSI snapshots support, I decided to clone the main repository, apply the same changes proposed by @blampe, (Makefile and Docker file to enabling building of multi-architecture docker images) and adapt the Github actions to generate my  my own multi-arch docker image (https://hub.docker.com/r/ricsanfre/velero-plugin-for-csi).

See arm64 branch in my cloned repository: https://github.com/ricsanfre/velero-plugin-for-csi/tree/arm64 and check the list of changes respect the main repository here: https://github.com/vmware-tanzu/velero-plugin-for-csi/compare/main...ricsanfre:velero-plugin-for-csi:arm64

In order to build also for `arm` architecture, you would need to change in the Makefile the `docker buildx` commands to add linux/arm in the list of platforms (--platform argument). I have not tested it since my cluster is only based on arm64 architecture. Maybe you can try it. Let me know if it works

Ricardo

**Comment by i5Js** [comment]
Hola Ricardo, 

Many thanks! I saw your fork, and I made ARM and ARM64 versions. I have the velero working fine, but I'm getting some errors. I used a different approach, and it works as a pod in K3s. Although I send the parameters using a config map, it doesn't seem to work the same way as before because it doesn't take them. 

I'll look into it. 

Hey, congrats! Everything looks amazing and you explain everything so well. 

BR

Javi

**Reply by ricsanfre** [reply]
Gracias

**Comment by i5Js** [comment]
Hola!

Everything works flawlessly now :) 

BR 

Javi
